From bd0b2aa8265109ab63e093a2c1d220f4d015e5e4 Mon Sep 17 00:00:00 2001
From: Pardha Saradhi K <pardha.saradhi.kesapragada@intel.com>
Date: Fri, 4 Mar 2016 12:11:52 +0530
Subject: [PATCH 1890/2508] ASoC: Intel: Skylake: Skip calling probe for HDA
 codecs

commit 3f35261af73467bec84a4af265a4106e022b6b31 from
https://github.com/01org/linux-apollolake-i

codec probe for HDA codecs like iDisp and HDMI increment the
runtime active counter. If the driver corresponding to the same
is not registered for any reason, the count remains 1 and
PM flows will not happen. To avoid this, function call to
such codec probe has been commented out, until a proper
fix is available to support HDA codecs

Change-Id: Ic09cb784f8f67a5fc2246463f99df550e5615235
Signed-off-by: Pardha Saradhi K <pardha.saradhi.kesapragada@intel.com>
Reviewed-on: https://git-gar-1.devtools.intel.com/gerrit/8335
Reviewed-by: Babu, Ramesh <ramesh.babu@intel.com>
Tested-by: Babu, Ramesh <ramesh.babu@intel.com>
---
 sound/soc/intel/skylake/skl.c |    9 +++++----
 1 files changed, 5 insertions(+), 4 deletions(-)

diff --git a/sound/soc/intel/skylake/skl.c b/sound/soc/intel/skylake/skl.c
index 5fbac90..2f826a5 100644
--- a/sound/soc/intel/skylake/skl.c
+++ b/sound/soc/intel/skylake/skl.c
@@ -690,10 +690,11 @@ static int skl_probe(struct pci_dev *pci,
 		goto out_dmic_free;
 
 	/* create codec instances */
-	err = skl_codec_create(ebus);
-	if (err < 0)
-		goto out_unregister;
-
+	if (IS_ENABLED(CONFIG_SND_SOC_HDAC_HDMI)) {
+		err = skl_codec_create(ebus);
+		if (err < 0)
+			goto out_unregister;
+	}
 	/* init debugfs */
 	skl->debugfs = skl_debugfs_init(skl);
 
-- 
1.7.5.4

